### PR TITLE
support initial notification configuration from front end

### PR DESCRIPTION
**Pull request by d-g-town**

### CODE REVIEW

File: api/server/handlers/notifications/get_notification_config.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+type GetNotificationConfigRequest struct{}
+
+// GetNotificationConfigResponse is the response object for the /notifications/{notification_config_id} endpoint
+type GetNotificationConfigResponse struct {
+	Config Config `json:"config"`
+}
+
+// ServeHTTP updates a notification config
+func (n *GetNotificationConfigHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctx, span := telemetry.NewSpan(r.Context(), "serve-notifications-update")
```
Comment (ferozemohideen) [inline api/server/handlers/notifications/get_notification_config.go]: span name

File: api/server/handlers/notifications/get_notification_config.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+	defer span.End()
+
+	project, _ := ctx.Value(types.ProjectScope).(*models.Project)
+
+	telemetry.WithAttributes(span,
```
Comment (ferozemohideen) [inline api/server/handlers/notifications/get_notification_config.go:52]: is this necessary?

File: api/server/handlers/notifications/get_notification_config.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+	}
+
+	response := &GetNotificationConfigResponse{
+		Config: configFromProto(ccpResp.Msg.Config),
+	}
+
+	n.WriteResult(w, r, response)
+}
+
+func configFromProto(proto *porterv1.NotificationConfig) Config {
```
Comment (ferozemohideen) [inline api/server/handlers/notifications/get_notification_config.go:98]: move to internal

File: api/server/handlers/notifications/get_notification_config.go [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+		mention = proto.SlackConfig.Mentions[0]
+	}
+
+	return Config{
+		Statuses: statuses,
+		Mention:  mention,
+	}
+}
+
+var transformProtoToStatusString = map[porterv1.EnumNotificationStatus]string{
```
Comment (ferozemohideen) [inline api/server/handlers/notifications/get_notification_config.go:119]: same with this

File: api/server/handlers/notifications/update_notification_config.go [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+	}
+
+	response := &UpdateNotificationConfigResponse{
+		ID: uint(updateResp.Msg.NotificationConfigId),
+	}
+
+	n.WriteResult(w, r, response)
+}
+
+func configToProto(config Config) *porterv1.NotificationConfig {
```
Comment (ferozemohideen) [inline api/server/handlers/notifications/update_notification_config.go:114]: internal

File: api/server/router/notification.go [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+		factory.GetResultWriter(),
+	)
+
+	routes = append(routes, &router.Route{
+		Endpoint: updateNotificationConfigEndpoint,
+		Handler:  updateNotificationConfigHandler,
+		Router:   r,
+	})
+
+	// GET /api/projects/{project_id}/notifications/{notification_config_id} -> notifications.NewNotificationConfigHandler
```
Comment (ferozemohideen) [inline api/server/router/notification.go:89]: NewGet

**Review by ferozemohideen — APPROVED** [review]

File: api/server/handlers/notifications/get_notification_config.go (hunk 2 above)
Comment (d-g-town) [reply api/server/handlers/notifications/get_notification_config.go]: no

**Pull request merged**